moveit_simple/src/moveit_simple.cpp

```diff
@@ -219,9 +218,70 @@ bool Robot::getJointSolution(const Eigen::Affine3d &pose, double timeout,
   return getIK(pose, local_seed, joint_point, timeout);
 }
 
+/*------------------------------------------------------------------------------------------------------------------------------*/
+
+/*ISSUE #20 -- getJointSolution() WITH CUSTOM TOOL FRAME FOR SOLVING IK*/
+bool Robot::getJointSolution(const Eigen::Affine3d &pose, const std::string& custom_tool_frame,
+                             double timeout, const std::vector<double> & seed,
+                             std::vector<double> & joint_point) const
+{
+
+  /*Transform Target/Goal Point from custom tool frame to moveit_end_link*/
+  const Eigen::Affine3d custom_frame_goal_pose = 
+                        transformToolToEndLink(pose, custom_tool_frame);
+
+  std::lock_guard<std::recursive_mutex> guard(m_);
+
+  std::vector<double> local_seed = seed;
+  if (seed.empty())
+  {
+    ROS_INFO_STREAM("Empty seed passed to getJointSolution, using current state");
+    local_seed =  getJointState();
+  }
+  return getIK(custom_frame_goal_pose, local_seed, joint_point, timeout);
+}
+
+/* Returns the goal pose transformed to moveit_end_link from Custom tool frame*/
+Eigen::Affine3d Robot::transformToolToEndLink(const Eigen::Affine3d &target_pose,
+                                              const std::string& custom_tool_frame) const
+{
+  
+  if(custom_tool_frame.compare(joint_group_->getEndEffectorName()) == 0 &&
+     robot_model_ptr_->hasLinkModel(custom_tool_frame)) {
+
+      ROS_INFO_STREAM("Custom Tool Frame is same as moveit end_link");
```

Comment:
The log message doesn't indicate what is happening...you should add something like "returning same target pose"